lib/atlas/graph_from_yaml.rb

```diff
@@ -30,31 +29,24 @@ def build_graph
     private
 
     def graph_nodes
-      Hash[@graph.nodes.map do |node|
-        attributes       = @graph_yaml.fetch(:nodes)[node.key] || {}
-        node_attributes  = attributes.slice!(:in, :out)
+      @graph.nodes.map do |node|
+        attributes = @graph_yaml.fetch(:nodes)[node.key] || {}
 
-        [ node, { slot: attributes, node: node_attributes } ]
-      end]
+        [ node,
+          attributes.except(:in, :out),
+          attributes.slice(:in, :out) ]
+      end
     end
 
     def update_slots(slots, attributes)
-      attributes.each_pair do |slot_key, share|
-        ref_slot = get_slot(slots, slot_key)
-
-        if share.is_a?(Numeric)
-          ref_slot.set(:share, share)
+      attributes.each_pair do |carrier, share_attributes|
+        ref_slot = if slots.include?(carrier)
+          slots.get(carrier)
         else
-          ref_slot.set(:type, share)
+          slots.add(carrier)
         end
-      end
-    end
 
-    def get_slot(slots, slot_key)
-      if slots.include?(slot_key)
-        slots.get(slot_key)
-      else
-        slots.add(slot_key)
+        ref_slot.properties = share_attributes
```

Comment:
Maybe better use `set()` in order to not delete properties which are missing in the yml